Always callback couchTableSource:self willUseCell:cell forRow:row

tleyden opened the pull request:
I had to make this change locally in my project when using CBLUiTableSource because when I added:

```
    func couchTableSource(source: CBUITableSource, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("GalleryTableViewCell", forIndexPath: indexPath) as! GalleryTableViewCell
        return cell
    }
```

this method:

```
    func couchTableSource(source: CBUITableSource, willUseCell cell: UITableViewCell, forRow row: CBLQueryRow) {
      ...
    }
```

was no longer being called back.  Making the above change fixed it, and I can't see how it would cause any side effects.

tleyden commented:
@pasin can you take a look?

pasin commented:
@tleyden Can you help me understand from your comment above why `CBUITableSource` not `CBLUITableSource`?

tleyden commented:
Oops, I meant `CBLUITableSource`, copy-paste error.

pasin commented on Source/API/CBLUITableSource.m:
```diff
@@ -170,23 +170,23 @@ - (UITableViewCell *)tableView:(UITableView *)tableView
     if (!cell) {
         // ...if it doesn't, create a cell for it:
-        cell = [tableView dequeueReusableCellWithIdentifier: @"CBLUITableDelegate"];
+        cell = [tableView dequeueReusableCellWithIdentifier: @"CBUITableDelegate"];
```
It should be `CBLUITableDelegate` ('L' missing).

pasin commented on Source/API/CBLUITableSource.m:
```diff
@@ -170,23 +170,23 @@ - (UITableViewCell *)tableView:(UITableView *)tableView
-        id delegate = _tableView.delegate;
-        if ([delegate respondsToSelector: @selector(couchTableSource:willUseCell:forRow:)])
-            [(id<CBLUITableDelegate>)delegate couchTableSource: self willUseCell: cell forRow: row];
+                                          reuseIdentifier: @"CBUITableDelegate"];
```
Same as above, 'L' is missing.

pasin commented on Source/API/CBLUITableSource.m:
```diff
@@ -170,23 +170,23 @@ - (UITableViewCell *)tableView:(UITableView *)tableView
+    // Allow the delegate to customize the cell:
+    id delegate = _tableView.delegate;
+    if ([delegate respondsToSelector: @selector(couchTableSource:willUseCell:forRow:)])
+        [(id<CBUITableDelegate>)delegate couchTableSource: self willUseCell: cell forRow: row];
```
Same as above, 'L' is missing.

pasin commented:
@tleyden If adding the missing 'L' character, I couldn't find the difference before and after change. Could you point me to the change that fixes the issue?

tleyden commented:
@pasin what happened was that I renamed things when I copied and pasted things into my own project, where I renamed `CBLUITable*` -> `CBUITable*`.  If you want I can fix these and re-push, or it might be easier if you just create a branch off of mine, make the changes, and then create a new PR and close this one.

pasin commented:
I can create the branch and fix it. Let me do this after 1.2.

snej commented:
By coincidence I just fixed the same problem as part of my commit 28fa0681. I hadn't even seen this PR!
